File: module_beauty/src/main/java/com/gas/beauty/main/mvp/MainContract.java
package com.gas.beauty.main.mvp;

import android.app.Activity;

import com.base.lib.mvp.IModel;
import com.base.lib.mvp.IView;
import com.gas.beauty.bean.GankBaseResponse;
import com.gas.beauty.bean.GankItemBean;

import java.util.List;

import io.reactivex.Observable;

public interface MainContract {


    interface View extends IView {
        Activity getActivity();
        void success();
        void onError();
    }


    interface Model extends IModel{
        Observable<GankBaseResponse<List<GankItemBean>>> getGirlList(int num, int page);
    }

}

File: module_beauty/src/main/java/com/gas/beauty/ui/main/mvp/MainContract.kt
package com.gas.beauty.ui.main.mvp

import android.app.Activity
import com.base.lib.mvp.IModel
import com.base.lib.mvp.IView
import com.gas.beauty.bean.GankResponse
import com.gas.beauty.bean.BeautyBean
import io.reactivex.Observable

interface MainContract {
    interface View : IView {
        val activity: Activity
        fun success()
        fun onError()
    }

    interface Model : IModel {
        fun getGirlList(num: Int, page: Int): Observable<GankResponse<List<BeautyBean>>>
    }
}